project-fortis-interfaces/src/actions/Admin/index.js

```diff
@@ -1,312 +1,345 @@
-          const error = 'Error, could not load trusted sources for admin page';
-          self.dispatch(constants.ADMIN.LOAD_FAIL, { error });
-        }
-      }))
-    }
-  },
-
-  remove_trusted_sources(sources) {
-    const self = this;
-    const dataStore = this.flux.stores.DataStore.dataStore;
-    if (!dataStore.loading) {
-      AdminServices.removeTrustedSources(sources, (err, response, body) => ResponseHandler(err, response, body, (error, graphqlResponse) => {
-        if (graphqlResponse && !error) {
-          const action = 'saved';
-          const trustedSourcesBeforeRemove = this.flux.stores.DataStore.dataStore.trustedSources;
-          const trustedSourcesRemoved = graphqlResponse.removeTrustedSources.sources
-          const trustedSourcesAfterRemove = getListAfterRemove(trustedSourcesBeforeRemove, trustedSourcesRemoved, 'rowKey');
-          self.dispatch(constants.ADMIN.LOAD_TRUSTED_SOURCES, {action});
-          self.dispatch(constants.DASHBOARD.LOAD_TRUSTED_SOURCES, {response: trustedSourcesAfterRemove})
-        } else {
-          const error = 'Error, could not load trusted sources for admin page';
-          self.dispatch(constants.ADMIN.LOAD_FAIL, { error });
-        }
-      }))
-    }
-  },
-
-    publish_events(events){
-        AdminServices.publishCustomEvents(events, (err, response, body) => ResponseHandler(err, response, body, (error, graphqlResponse) => {
-            let action = 'saved';
-            const self = this;
-
-            if (graphqlResponse && !error) {
-                self.dispatch(constants.ADMIN.PUBLISHED_EVENTS, {action});
-            }else{
-                action = 'failed';
-                console.error(`[${error}] occured while processing message request`);
-                self.dispatch(constants.ADMIN.PUBLISHED_EVENTS, {action});
-            }
-        }));
-    }
-};
-
-const methods = { ADMIN: _methods };
-
-module.exports = {
-    constants,
-    methods
-};
+import { SERVICES as AdminServices } from '../../services/Admin';
+import { ResponseHandler } from '../shared';
+import { doNothing } from '../../utils/Utils';
+import constants from '../constants';
+import differenceBy from 'lodash/differenceBy';
+import differenceWith from 'lodash/differenceWith';
+import some from 'lodash/some';
+
+function getListAfterRemove(listBeforeRemove, itemsRemoved, keyBy) {
+  const customComparator = (a, b) => {
+    function isPropertyEqual(property) {
+      return a[property] === b[property];
+    }
+    return keyBy.every(isPropertyEqual);
+  }
+
+  if (keyBy.constructor === Array) return differenceWith(listBeforeRemove, itemsRemoved, customComparator);
+  else return differenceBy(listBeforeRemove, itemsRemoved, keyBy);
+}
+
+function addIdsToUsersForGrid(users) {
+  users.forEach(user => user.id = `${user.identifier}-${user.role}`);
+}
+
+function prepareBlacklistForGrid(rows) {
+  rows.forEach(row => {
+    if (row && row.filteredTerms && row.filteredTerms.constructor === Array) {

```

Comment:
NB: it might be clearer/less hassle to use lodash's [isArray](https://lodash.com/docs/4.17.5#isArray) to check for array type.

```
import isArray from 'lodash/isArray';

// ...

if (row && isArray(row.filteredTerms)) {
  // ...
}
```

Same for Array check on L17 above.